File: problems/001/001.rb
def solution
  sum = 0

  (0..999).each do |i|
    sum += i if i % 3 == 0 || i % 5 == 0
  end

  sum
end

# XXX Shorter but 60% slower
#def solution
  #(0..999).select { |x| x % 3 == 0 || x % 5 == 0 }.inject(:+)
#end

if ARGV.length == 1 && ARGV[0] == '-a'
  puts solution
  exit
end

STDOUT.sync = true
STDIN.each do |line|
  iters = line.to_i

  start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
  (1..iters).each do ||
    solution
  end
  end_ = Process.clock_gettime(Process::CLOCK_MONOTONIC)

  elapsed = ((end_ - start) * 1e9).to_i

  puts elapsed
end

File: problems/001/001.jl
function solution()
  sum = 0

  for x = 1:999
    if x % 3 == 0 || x % 5 == 0
      sum += x
    end
  end

  sum
end

# XXX Shorter, but 100%+ slower. std_dev is way higher as well
#=function solution()=#
  #=sum([x % 3 == 0 || x % 5 == 0 ? x : 0 for x = 0:999])=#
#=end=#

if length(ARGS) == 1 && ARGS[1] == "-a"
  println(solution())
  exit(0)
end

for line in eachline(STDIN)
  iters = int(line)

  start = time_ns()
  for i in 1:iters
    solution()
  end
  end_ = time_ns()

  println(end_ - start)
end
